daemon/daemon.go

```diff
@@ -1024,6 +1031,14 @@ func (daemon *Daemon) Reload(config *Config) error {
 	// We emit daemon reload event here with updatable configurations
 	attributes["debug"] = fmt.Sprintf("%t", daemon.configStore.Debug)
 	attributes["live-restore"] = fmt.Sprintf("%t", daemon.configStore.LiveRestoreEnabled)
+
+	if daemon.configStore.InsecureRegistries != nil {
+		insecureRegistries, _ := json.Marshal(daemon.configStore.InsecureRegistries)
```

Comment:
I'm not sure how it's in other place, but it looks quite bad to ignore errors from `Marshal`.